### PR TITLE
Fix schema info serialization

### PR DESCRIPTION
**Pull request by jaredmoo**

### CODE REVIEW

File: Src/ElasticScale.Client/ShardManagement/Scripts/UpgradeShardMapManagerGlobalFrom1.2To1.3.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+					cast(SchemaInfo as nvarchar(max)), 
+					'<_referenceTableSet i:type="ArrayOfReferenceTableInfo">', 
+					'<ReferenceTableSet i:type="ArrayOfReferenceTableInfo">'),
+				'</_referenceTableSet>',
+				'</ReferenceTableSet>'),
+		'<_shardedTableSet i:type="ArrayOfShardedTableInfo">', 
+		'<ShardedTableSet i:type="ArrayOfShardedTableInfo">'),
+	'</_shardedTableSet>',
+	'</ShardedTableSet>')
+go
```
Comment (apurvans) [inline Src/ElasticScale.Client/ShardManagement/Scripts/UpgradeShardMapManagerGlobalFrom1.2To1.3.sql]: Update persisted version information at the end of script -
-- update version as 1.3
update
	__ShardManagement.ShardMapManagerGlobal 
set 
	StoreVersionMinor = 3
where
	StoreVersionMajor = 1 and StoreVersionMinor = 2

File: Test/ElasticScale.ShardManagement.UnitTests/ShardMapManagerUpgradeTests.cs [hunk 2]
```diff
@@ -1,339 +1,448 @@
...
+            smm.UpgradeGlobalStore(new Version(1, 1));
+
+            // Below call should succeed as latest supported major version of library matches major version of deployed store.
+            ShardMap sm = smm.CreateListShardMap<int>(ShardMapManagerUpgradeTests.s_shardMapNames[0]);
+            Assert.IsNotNull(sm);
+
+            // Upgrade to version 1.2
+            smm.UpgradeGlobalStore(new Version(1, 2));
+
+            // Upgrade to latest version (1.2): no-op
```
Comment (apurvans) [inline Test/ElasticScale.ShardManagement.UnitTests/ShardMapManagerUpgradeTests.cs]: This is not a no-op as we have v1.3 now.

File: Test/ElasticScale.ShardManagement.UnitTests/ShardMapManagerUpgradeTests.cs [hunk 3]
```diff
@@ -1,339 +1,448 @@
...
+        }
+
+        [TestMethod]
+        public void TestSerializationFixInVersion1_3()
+        {
+            // Get shard map manager
+            ShardMapManager smm = ShardMapManagerFactory.GetSqlShardMapManager(
+                Globals.ShardMapManagerConnectionString,
+                ShardMapManagerLoadPolicy.Lazy);
+
```
Comment (apurvans) [inline Test/ElasticScale.ShardManagement.UnitTests/ShardMapManagerUpgradeTests.cs]: smm.UpgradeGlobalStore(new Version(1, 2)) before proceeding. That way we can verify that issue exists in v1.2 and then got fixed in v1.3. This particular fix is simpler, but if we ever have a regression 'in-cycle' then starting from fixed version will catch it correctly.

**Comment by jaredmoo** [comment]
Closing pull request. This approach doesn't work, because a v1.1.0 client could come back after the upgrade and re-insert data with the wrong serialization format.

**Pull request closed without merging**